tests/test_web.py

```diff
@@ -199,3 +200,46 @@ def test_smtp_starttls_env_flag_must_be_true_boolean(environment, http, server,
     http.post("/invoice", json=email_invoice_request_body)
 
     assert server.starttls.call_count == 0
+
+
+@pytest.mark.parametrize(
+    "environment",
+    [
+        (
+            {
+                "INVOICE_UTILS_MAIL_SUBJECT": "test subject",
+                "INVOICE_UTILS_BODY_TEMPLATE_PATH": "test_template.html",
+                "INVOICE_UTILS_SENDER_EMAIL": "test@email.com"
+            }
+        ),
+        (
+            {
+                "INVOICE_UTILS_MAIL_SUBJECT": "test subject",
+                "INVOICE_UTILS_BODY_TEMPLATE_PATH": DEFAULT_BODY_TEMPLATE_PATH,
+                "INVOICE_UTILS_SENDER_EMAIL": "test@email.com"
+            }
+        )
+    ],
+    indirect=["environment"]
+)
+def test_email_body_was_sent_with_expected_body(
+        environment, http, server, email_invoice_request_body
+):
+    http.post("/invoice", json=email_invoice_request_body)
+
+    env = Environment(
```

Comment:
The test knows too much about the template's logic. Ideally, the test just calls the endpoint and if a template is specified, the e-mail contains an expected string. There shouldn't be any template manipulation in the test.